File: src/main/java/net/jcflorezr/exceptions/AudioFileLocationException.java
package net.jcflorezr.exceptions;

import java.nio.file.Path;

public class AudioFileLocationException extends BadRequestException {

    public AudioFileLocationException(String message, String suggestion) {
        super(message, suggestion);
    }

    public static AudioFileLocationException audioFileDoesNotExist(Path audioFileName) {
        return new AudioFileLocationException("The audio file '" + audioFileName + "' does not exist.", null);
    }

    public static AudioFileLocationException audioFileShouldNotBeDirectory(Path audioFileName) {
        return new AudioFileLocationException("'" + audioFileName + "' should be a file, not a directory.", null);
    }

    public static AudioFileLocationException outputDirectoryDoesNotExist(Path outputAudioClipsPath) {
        return new AudioFileLocationException("The directory '" + outputAudioClipsPath + "' does not exist.", null);
    }

    public static AudioFileLocationException sameAudioFileAndOutputDirectoryLocation() {
        return new AudioFileLocationException("The audio file location cannot be the same as the " +
                "output audio clips location.", null);
    }

    public static AudioFileLocationException emptyAudioFileLocationObject() {
        String suggestion = "Example of body request: {\"audioFileName\": \"/any-audio-file.wav\", \"outputAudioClipsDirectoryPath\": \"/any-output-directory/\"}";
        return new AudioFileLocationException("There is no body in the current request.", suggestion);
    }

    public static AudioFileLocationException mandatoryFieldsException() {
        String suggestion = "Mandatory fields are: [audioFileName, outputAudioClipsDirectoryPath]";
        return new AudioFileLocationException("There are empty mandatory fields.", suggestion);
    }
}

File: src/main/kotlin/net/jcflorezr/exception/AudioFileLocationException.kt
package net.jcflorezr.exception;

class AudioFileLocationException(
        message: String,
        suggestion: String? = null
) : BadRequestException(message, suggestion) {
    companion object {

        fun audioFileDoesNotExist(audioFileName: String): AudioFileLocationException {
            return AudioFileLocationException(
                    message = "The audio file '$audioFileName' does not exist."
            )
        }

        fun audioFileShouldNotBeDirectory(audioFileName: String): AudioFileLocationException {
            return AudioFileLocationException(
                    message = "'$audioFileName' should be a file, not a directory."
            )
        }

        fun outputDirectoryDoesNotExist(outputAudioClipsPath: String): AudioFileLocationException {
            return AudioFileLocationException(
                    message = "The directory '$outputAudioClipsPath' does not exist."
            )
        }

        fun sameAudioFileAndOutputDirectoryLocation(): AudioFileLocationException {
            return AudioFileLocationException(
                    message = "The audio file location cannot be the same as the " + "output audio clips location."
            )
        }

        fun emptyAudioFileLocationObject(): AudioFileLocationException {
            return AudioFileLocationException(
                    message = "There is no body in the current endpoint.",
                    suggestion = "Example of body endpoint: {\"audioFileName\": \"/any-audio-file.wav\", \"outputAudioClipsDirectoryPath\": \"/any-output-directory/\"}"
            )
        }

        fun mandatoryFieldsMissingException(): AudioFileLocationException {
            return AudioFileLocationException(
                    message = "There are empty mandatory fields.",
                    suggestion = "Mandatory fields are: [audioFileName, outputAudioClipsDirectoryPath]"
            )
        }
    }
}
